### PR TITLE
EVG-7106: backup collection job

### PR DESCRIPTION
**Pull request by tychoish**
I've been editing the PR to add more collection, here's the state of the collections in the last PR: 

## candidates for fullbackup, not presently included
- build_baron_notes
- feedback
- json
- artifact_files

## we can probably indexonly these backups, not presently included
- alertrecord
- alerts
- notifications
- patchfiles.chunks
- patchfiles.files
- process_runtimes
- task_bk

## we should probably save the last day or week worth of these documents, not presently included
builds
parser_projects
hosts
task_bk
patches
tasks
tickets
versions
volumes

## backup all documents in these collections
- github_hooks
- project_aliases
- project_ref
- project_vars
- admin
- distro
- pushes
- repo_revisions
- globals
- credentials
- users
- keyval_plugin
- commit_queue
- manifest

## only backup indexes:
- old_tasks
- daily_stats_status
- daily_task_stats
- daily_test_stats
- hourly_test_stats
- task_alias_queues
- task_queues
- test_logs
- testresults
- notify_history
- notify_times
- event_log
- subscriptions
- patch_intents

## collections in the db that aren't referenced in the app
- host_bk
- agg_temp
- waittimes
- task_event_log
- spawn_requests
- ola
- patches_backup
- lock

### CODE REVIEW

**Comment by john-m-liu** [comment]
What's the benefit of this over mms backups?

File: self-tests.yml [hunk 1]
```diff
@@ -517,7 +517,7 @@ buildvariants:
     expansions:
       goos: linux
       goarch: amd64
-      gobin: /opt/golang/go1.10/bin/go
+      gobin: /opt/golang/go1.10/bin/gogd
```
Comment (bsamek) [inline self-tests.yml]: typo

File: units/backup_spec.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	util.RoundPartOfDay(6)
+	queue, err := env.RemoteQueueGroup().Get(ctx, "backup_collector")
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	settings := env.Settings()
+
+	collections := appendAmboyCollections(settings.Amboy, []backup.Options{})
+	collections = appendFullBackupCollections(settings.Database.DB, collections)
+	collections = appendInexOnlyBackupCollections(settings.Database.DB, collections)
```
Comment (bsamek) [inline units/backup_spec.go]: typo Index

File: units/backup_spec.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	"github.com/pkg/errors"
+)
+
+func AddBackupJobs(ctx context.Context, env evergreen.Environment, ts time.Time) error {
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if flags.DRBackupDisabled {
```
Comment (bsamek) [inline units/backup_spec.go]: These should also noop if BackupConfig isn't fully populated.

**Pull request merged**